### PR TITLE
Salesforce Connector Additional Configuration Topics

### PR DESCRIPTION
**Pull request by dhirsch1**

### CODE REVIEW

File: modules/ROOT/pages/salesforce/salesforce-connector-100-config-topics.adoc [hunk 1]
```diff
@@ -2,67 +2,54 @@
...
-
-Each event that travels through your flows contains information about the Salesforce data that has changed, how it changes, and when. The connector parses this information and sends you information that a flow can work with.
-
-Inbound properties of events:
+Your app can receive events by subscribing to an existing Salesforce topic or by creating a new topic and then subscribing to it.
 
-* `payload`
-* `createdDate`
-* `replayId`
+Each event that travels through your flow contains information about the Salesforce data that changed, including how the data changed and when the change occurred. The connector parses this information and sends your app the information.
```
Comment (alexandruatomei9) [inline modules/ROOT/pages/salesforce/salesforce-connector-100-config-topics.adoc]: I think here the phrasing is not right. _The connector parses this information and sends your app the information._ can be removed.

File: modules/ROOT/pages/salesforce/salesforce-connector-100-config-topics.adoc [hunk 2]
```diff
@@ -2,67 +2,54 @@
...
 
-Salesforce stores events for 24 hours, so you can retrieve stored events during that retention window. A subscriber (to a topic or channel) can retrieve events at any time and is not restricted to listening to events at the time they are sent.
+Salesforce stores events for 24 hours. You can retrieve stored events during that retention window. A subscriber to a topic or channel can retrieve events at any time and is not restricted to listening to events at the time they are sent.
 
-Each broadcast event is assigned a numeric ID. IDs are incremented and not guaranteed to be contiguous for consecutive events. Each ID is guaranteed to be higher than the ID of the previous event. For example, the event following the event with ID 999 can have an ID of 1025. The ID is unique for the organization and the channel. The IDs of deleted events are not reused.
+*Reviewers: the above paragraph is not clear. Does the subscriber have to retrieve events within 24 hours?*
```
Comment (alexandruatomei9) [inline modules/ROOT/pages/salesforce/salesforce-connector-100-config-topics.adoc]: The idea is that a user can create a topic for accounts, for example. He does not subscribe to that topic immediately. He creates or updates accounts and these events are pushed on the topic he created - these events live on the topic for 24 hours, if he subscribes during these 24 hours he will be able to receive all the events. If the 24 hours period passes he won't be able to receive those events anymore but he will receive newer events(that are not yet expired = that occurred in the past 24 hours in relation with the subscription time)

File: modules/ROOT/pages/salesforce/salesforce-connector-100-config-topics.adoc [hunk 3]
```diff
@@ -128,34 +117,41 @@ In this XML example, the Replay Topic Listener acts like an inbound endpoint for
...
+You can use Salesforce Connector as an inbound connector to stream data from Salesforce into your application. To use the connector in this capacity, place Salesforce Connector at the start of your flow and do not wrap it in a poll scope.
 
-NOTE: Studio automatically converts the connector to Salesforce (Streaming) mode. Technically, this is still the same connector, but it accesses the Salesforce Streaming API. This means that the only operation the converted connector can perform is *Subscribe Topic Listener* (that is, subscribe to PushTopic).
+Studio automatically converts the connector to Salesforce Streaming mode, which accesses the Salesforce Streaming API. The converted connector can perform only the  *Subscribe Topic Listener* operation in this mode.
 
-image::salesforce/salesforce-studio-subscribe-streaming-channel.png[subscribe streaming channel]
+*Reviewers: Is *Subscribe Topic Listener* correct in the paragraph aboe=ve, or should this be *Subscribe Channel Listener?*
```
Comment (alexandruatomei9) [inline modules/ROOT/pages/salesforce/salesforce-connector-100-config-topics.adoc]: This is a bit outdated - meaning that in mule 4 you don't drop the connector at the start of the flow, but rather you pick one of the Sources: Subscribe Topic, Subscribe Channel, Replay Topic or Replay Channel.  - these are the "operations" that the "converted" connector can perform.

File: modules/ROOT/pages/salesforce/salesforce-connector-100-config-topics.adoc [hunk 4]
```diff
@@ -128,34 +117,41 @@ In this XML example, the Replay Topic Listener acts like an inbound endpoint for
...
+Follow these steps to create a streaming channel:
+
+. Log in to your Salesforce Developer Edition organization.
 . Under *All Tabs (+)*, select *Streaming Channels*.
 . On the *Streaming Channels* tab, select *New* to create a new streaming channel.
-. Enter `/u/notifications/ExampleUserChannel` in the *Streaming Channel Name* field, and an optional description.
+. Enter `/u/notifications/ExampleUserChannel` in the *Streaming Channel Name* field.
+. Enter an optional description.
+
+*Reviewers: Where does the next paragraph fit in? Is it a step to perform after step 5 in the previous procedure?*
```
Comment (alexandruatomei9) [inline modules/ROOT/pages/salesforce/salesforce-connector-100-config-topics.adoc]: The steps above describe how to create a streaming channel in the Salesforce platform. The text below is a note saying that one can also use the connector to create the streaming channel if he wants, by using one of the 2 operations Create or Publish Streaming Channel

File: modules/ROOT/pages/salesforce/salesforce-connector-100-config-topics.adoc [hunk 5]
```diff
@@ -179,28 +174,25 @@ After you create a streaming channel, you can start receiving events by subscrib
...
-* `type` - Maps to the Type JSON property in data.
-* `createdDate` - Maps to the createdDate JSON property in data.
+* `channel` maps to the Channel JSON property.
+* `type` maps to the Type JSON property in data.
+* `createdDate` maps to the createdDate JSON property in data.
 
 Except for `channel`, each property inside an event is available as an inbound property.
 
-=== Replay Events from a Streaming Channel
+// This does not make sense.
```
Comment (alexandruatomei9) [inline modules/ROOT/pages/salesforce/salesforce-connector-100-config-topics.adoc]: What do you mean? It makes sense you can either Subscribe to a channel or Replay the events of a channel

File: modules/ROOT/pages/salesforce/salesforce-connector-100-config-topics.adoc [hunk 6]
```diff
@@ -266,33 +257,23 @@ The JSON received as a response from the push event operation looks something li
...
 
-The Salesforce Bulk API loads batches of your organization's data into Salesforce.
+The Salesforce Bulk API loads batches of your organization's data into Salesforce. Salesforce Connector provides the Create and Create Bulk operations for working with the Bulk API.
 
-The Salesforce connector provides the Create and Create Bulk operations for working
-with the Bulk API.
+For all bulk operations, Salesforce handles the creation process in the background. The connector replies with a BatchInfo object, which contains the ID of the batch and the ID of the job it creates to upload the objects.
 
-For all bulk operations, Salesforce handles the creation process in the background, so the connector does not reply with a collection of SaveResults because it does not have them yet. Instead, the connector replies with a BatchInfo object, which contains the ID of the batch and the ID of the job it creates to upload those objects.
+// Find out which objects this refers to
```
Comment (alexandruatomei9) [inline modules/ROOT/pages/salesforce/salesforce-connector-100-config-topics.adoc]: Any type of Salesforce entity: Account, Contact, Lead and all the other available - even custom objects

File: modules/ROOT/pages/salesforce/salesforce-connector-100-config-topics.adoc [hunk 7]
```diff
@@ -318,7 +299,7 @@ output application/java
 
 === Fields To Null
 
-The configurations have a checkbox called Can Clear Fields by Updating Field Value to Null. If checked, all the fields in a request that have a Null value are added to the `fieldsToNull` field and sent to Salesforce.
+The configurations have a checkbox called *Can Clear Fields by Updating Field Value to Null*. If checked, all the fields in a request that have a Null value are added to the `fieldsToNull` field and sent to Salesforce.
```
Comment (alexandruatomei9) [inline modules/ROOT/pages/salesforce/salesforce-connector-100-config-topics.adoc]: This whole section should be removed(Fields to null) as it does not apply anymore for Salesforce 10.

File: modules/ROOT/pages/salesforce/salesforce-connector-100-config-topics.adoc [hunk 8]
```diff
@@ -328,19 +309,17 @@ Unless you configure the External ID Field Name for the sObject to which you are
 
 The upsert operation does not work with the sObject `priceBookentry2`.
 
-While you can't change the `contentType` for bulk upsert, you can use the `Create Job` operation to set the contentType to either CSV or zipped CSV (if you're near the character limit). Follow up with the `Create Batch` operation.
+A;through you can't change the `contentType` for bulk upsert, you can use the `Create Job` operation to set the contentType to either CSV or zipped CSV (if you're near the character limit). Follow up with the `Create Batch` operation.
```
Comment (alexandruatomei9) [inline modules/ROOT/pages/salesforce/salesforce-connector-100-config-topics.adoc]: Although

File: modules/ROOT/pages/salesforce/salesforce-connector-100-config-topics.adoc (hunk 6 above)
Comment (dhirsch1) [reply modules/ROOT/pages/salesforce/salesforce-connector-100-config-topics.adoc]: Used the term "batched objects" to make this clearer.

**Comment by dhirsch1** [comment]
The Additional Configuration Topic is included in PR 756.

**Pull request closed without merging**